src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -551,11 +551,12 @@ export const GlobalProvider = memo(
         );
 
         const createNode = useCallback(
-            (proto: NodeProto): void => {
+            (proto: NodeProto, parentId?: string): void => {
                 changeNodes((nodes) => {
-                    const parent = hoveredNode
-                        ? nodes.find((n) => n.id === hoveredNode)
-                        : undefined;
+                    const parent =
+                        parentId || hoveredNode
+                            ? nodes.find((n) => n.id === (parentId ?? hoveredNode))
+                            : undefined;
```

Comment:
[I like my code how I like my beer](https://www.youtube.com/watch?v=yKPA5l9pvFo).

```suggestion
                    const searchId = parentId ?? hoveredNode;
                    const parent =
                        searchId
                            ? nodes.find((n) => n.id === searchId)
                            : undefined;
```